src/youtube_chapters_demo.py

```diff
@@ -0,0 +1,177 @@
+#!/usr/bin/env python
+
+import os
+import sys
+import youtube_dl
+import re
+import datetime
+
+import opentimelineio as otio
+
+
+# Input: @description_file contains the Youtube description
+# Function should parse the Youtube description to find any table-of-contents entries. 
+# The table-of-contents entries will contain a timestamp paired with a chapter title. 
+# For example: (00:12:15) Thor fights Thanos
+# 
+# The function should return an array of all such timestamps and chapter titles. 
+#
+def process_youtube_description(description_file): 
+  with open(description_file) as f:
+    lines = f.readlines()
+
+  # Should match the following timestamp patterns: 
+  # 01:34:21 chapter title
+  # 1:34:21 chapter title 
+  # 34:21 chapter title
+  # 4:21 chapter title
+  # (4:21) chapter title
+  # [4:21] chapter title
+
+
+  pattern = re.compile('((?:\d+:)+\d{2}).*\s(.+)')
+  chapters = []
+
+  for line in lines: 
+      matches = pattern.findall(line) 
+      if len(matches) > 0: 
```

Comment:
the truthiness of `matches` is good enough:
```suggestion
      if matches: 
```